include/swift/RemoteInspection/ReflectionContext.h

```diff
@@ -1202,8 +1202,41 @@ class ReflectionContext
     }
   }
 
-  const RecordTypeInfo *getRecordTypeInfo(const TypeRef *TR,
-                              remote::TypeInfoProvider *ExternalTypeInfo) {
+  /// Given a typeref, attempt to calculate the unaligned start of this
+  /// instance's fields. For example, for a type without a superclass, the start
+  /// of the instance fields would after the word for the isa pointer and the
+  /// word for the refcount field. For a subclass the start would be the after
+  /// the superclass's fields. For a version of this function that performs the
+  /// same job but starting out with an instance pointer check
+  /// MetadataReader::readInstanceStartFromClassMetadata.
+  llvm::Optional<unsigned>
+  readUnalignedFieldStartOffsetFromTypeRef(const TypeRef *TR) {
```

Comment:
Nit: Does it read it or calculate it?
Maybe computeUnalined... or getUnaligned...?
The FromTypeRef suffix is a bit redundant here because you're passing a TR as the parameter. In `readInstanceStartFromClassMetadata` it makes more sense because it explains what kind of metadata pointer the parameter is.